internal/compute/provider.go

```diff
@@ -55,6 +44,62 @@ func NewComputeProvider(provider models.ProviderID) (Provider, error) {
 }
 
 // NewProvisioner creates a new system provisioner
-func NewProvisioner(jobID string) Provisioner {
-	return NewAnsibleConfigurator(jobID)
+func NewProvisioner(jobID string) provisioner.Provisioner {
+	// Extract job name from job ID (format: "job-20250411-134559")
+	jobName := strings.TrimPrefix(jobID, "job-")
```

Comment:
[non-blocking] as a default we should avoid string manipulations in higher level packages. This should be a `Name()` method on the `Job` model.  If we don't have the job model when calling functions like this, then we could also have a `JobID` type which has a `Name` method that does the string manipulation. 

this helps reduce the surface error for bugs, so we aren't doing this manipulations in multiple places.